src/vdom/component.js

```diff
@@ -21,11 +21,18 @@ export function setComponentProps(component, props, opts, context, mountAll) {
 	if ((component.__key = props.key)) delete props.key;
 
 	if (!component.base || mountAll) {
-		if (component.componentWillMount) component.componentWillMount();
+		if (component.componentWillMount) {
+			options.warn("'componentWillMount' is deprecated");
```

Comment:
I'd much rather see these warnings in our devtools (`debug/index.js`). Strings contribute quite a bit to our file size and moving them there would prevent bloating core.

Reply:
Sure. Didn't think of that. Anyway I do not know what should be deprecated yet :)